cmd/web-handlers.go

```diff
@@ -351,27 +351,65 @@ func (web *webAPIHandlers) SetAuth(r *http.Request, args *SetAuthArgs, reply *Se
 	if !isValidSecretKey.MatchString(args.SecretKey) {
 		return &json2.Error{Message: "Invalid Secret Key"}
 	}
+
 	cred := credential{args.AccessKey, args.SecretKey}
+
+	// Helper function - in case of an error in addition to a peer
+	// error, we return both together.
+	appendErr := func(peerErr, err *json2.Error) *json2.Error {
+		if peerErr == nil {
+			return err
+		}
+		return &json2.Error{
+			Message: "One or more errors happened!",
+			Data:    []*json2.Error{peerErr, err},
+		}
+	}
+
+	// Notify all Minio Peers to update credentials
+	errsMap := updateCredsOnPeers(cred)
+	var peerErr *json2.Error
+	if len(errsMap) != 0 {
+		peerErr = &json2.Error{
+			Message: "Could not update credentials on one or more peers.",
+			Data:    errsMap,
+		}
+	}
+
```

Comment:
It is somewhat arbitrary, as we cannot say how many peers may succeed in updating the credentials. If there are one or more failures, there is a partition - nodes with the new password and nodes with the old. Nodes in the same partition may communicate with each other but not with nodes outside the partition.

Here we are just being optimistic and trying to maximize the number of servers where the password change takes place. At the end we will display a message to the user saying on what servers the update could not be made.